irods/connection.py

```diff
@@ -754,3 +756,5 @@ def temp_password(self):
         # Convert and return answer
         msg = response.get_main_message(GetTempPasswordOut)
         return obf.create_temp_password(msg.stringToHashWith, self.account.password)
+
+import irods.auth
```

Comment:
Is there a reason to move this import to the bottom of the script? I'm assuming we want something there to be executed/evaluated later, but asking to be sure.

Reply:
I believe there's a cyclic import dependency somewhere above it in the module.  I didn't think very hard about it or try to track it down, I admit.  Placing this `irods.auth` import with the others in the file header doesn't work, but placing it at the end does.

Reply:
I think I see it now. irods.auth imports irods.session,  which needs irods.pool, which needs irods.connection... Guess I could move it inside the static method where it's used, I just think this to be simpler.

Reply:
I'll place it back with the others at the head of the file. It should fail in testing if it truly creates a cyclic dependency.

Reply:
As it turned out, depending on order of import, it was a problem, so I moved the import inside Connection constructor, where it never incurs any import overhead unless it's the first time the module is imported.  (Even that is unlikely, and I'd say impossible in most use cases)